microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+            modinfo = node.tools[Modinfo]
+            for module in hv_modules:
+                module_version = VersionInfo.parse(modinfo.get_version(module))
+                assert_that(module_version).described_as(
+                    f"Version of {module} does not match LIS version"
+                ).is_equal_to(lis_version)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the presence of all Hyper V drivers using lsmod
+           to look for the drivers not directly loaded into the kernel.
+        """,
+        priority=0,
+    )
+    def verify_hyperv_modules(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        hv_modules = self.__get_modules_not_in_kernel(node)
+        distro_version = node.os.information.version
+
+        # Some versions of RHEL and CentOS have the LIS package installed
+        #   which includes extra drivers
+        if isinstance(node.os, Redhat):
+            modprobe = node.tools[Modprobe]
+            lis_installed = node.os.package_exists("microsoft-hyper-v")
+
+            if lis_installed:
+                hv_modules.append("pci_hyperv")
+                modprobe.run("pci_hyperv", sudo=True)
+
+            if (
+                distro_version >= "7.3.0" or distro_version < "7.5.0"
+            ) and lis_installed:
+                hv_modules.append("mlx4_en")
+                modprobe.run("mlx4_en", sudo=True)
+
+        # Counts the Hyper V drivers loaded as modules
+        num_modules_loaded = 0
+        lsmod = node.tools[Lsmod]
+        for module in hv_modules:
+            if lsmod.module_exists(module):
+                log.info(f"Module {module} present")
+                num_modules_loaded = num_modules_loaded + 1
+            else:
+                log.error(f"Module {module} absent")
+
+        num_modules_expected = len(hv_modules)
+        assert_that(num_modules_loaded).described_as(
+            "Not all Hyper V drivers are present."
+        ).is_equal_to(num_modules_expected)
+
+    def __get_modules_not_in_kernel(self, node: RemoteNode) -> list[str]:
+        """
+        Returns the hv_modules that are not directly loaded into the kernel.
+        """
+        hv_modules_configuration = {
+            "hv_storvsc": "CONFIG_HYPERV_STORAGE",
+            "hv_netvsc": "CONFIG_HYPERV_NET",
+            "hv_vmbus": "CONFIG_HYPERV",
+            "hv_utils": "CONFIG_HYPERV_UTILS",
+            "hid_hyperv": "CONFIG_HID_HYPERV_MOUSE",
+            "hv_balloon": "CONFIG_HYPERV_BALLOON",
+            "hyperv_keyboard": "CONFIG_HYPERV_KEYBOARD",
+        }
+        uname = node.tools[Uname]
+        kernel_version = uname.get_linux_information().kernel_version_raw
+        config_path = f"/boot/config-{kernel_version}"
+
+        for module in hv_modules_configuration:
+            if (
+                node.execute(
+                    f"grep {hv_modules_configuration[module]}=y {config_path}"
```

Comment:
1. They can be added to a list, if the line is `=m`. Rather than `y` or `m`, it may be other values.
2. Match it from beginning of the line. We found a bug that the line is commented out with a `#`. So, it's worth to validate the line is effective without any prefix.

Reply:
The idea is that there is a list of drivers expected to be present: hv_storvsc, hv_netvsc, hv_vmbus, hv_utils, hid_hyperv, hv_balloon, hyperv_keyboard. We then must skip any drivers that are loaded directly with `y` because they will not show up in lsmod, and their version from modinfo is not relevant. It would not make sense to only check for `m` because the driver could be left out entirely. This is also my motivation for naming the function `_get_expected_modules` because it returns the modules we expect to see in lsmod, but I can continue to think of a better name.